ui/mods/msu/mod_settings/color_picker_setting.js

```diff
@@ -0,0 +1,156 @@
+
+	// Tooltip
+	this.title.bindTooltip({ contentType: 'ui-element', elementId: "msu-settings." + _mod.id + "." + _setting.id });
+	this.button.bindTooltip({ contentType: 'ui-element', elementId: "msu-settings." + _mod.id + "." + _setting.id });
+};
+
+ColorPickerSetting.prototype.createColorPickerPopup = function ()
+{
+	var self = this;
+	this.popup = this.parent.createPopupDialog('Color Picker', null, null, 'color-picker-popup', false);
+	this.popup.values = {
+		"Red" : 0,
+		"Green" : 0,
+		"Blue" : 0,
+		"Alpha" : 0
+	};
+	this.cloneSettingsToFrom(this.popup.values, this.values);
+	this.popup.addPopupDialogOkButton(function (_dialog){
+		self.cloneSettingsToFrom(self.values, self.popup.values);
+		self.updateColor();
+		_dialog.destroyPopupDialog();
+	})
+	this.popup.addPopupDialogCancelButton(function (_dialog){
+		_dialog.destroyPopupDialog();
+	})
+	this.popup.addPopupDialogContent(this.createPopupContent())
+	this.updateColorInPopup();
+
+};
+
+ColorPickerSetting.prototype.createPopupContent = function ()
+{
+	var result = $('<div class="pick-color-content-container"/>');
+	this.popup.currentColorDisplay = $('<div class="current-color-display"/>');
+	result.append(this.popup.currentColorDisplay);
+	this.createColorRowContent(result, "Red");
+	this.createColorRowContent(result, "Green");
+	this.createColorRowContent(result, "Blue");
+	this.createColorRowContent(result, "Alpha");
+	return result
+}
+
+ColorPickerSetting.prototype.createColorRowContent = function(_parentDiv, _name )
+{
+	var self = this;
+	var result = $('<div class="row"/>');
+	_parentDiv.append(result);
+
+	result.title = $('<div class="title title-font-big font-bold font-color-title">' + _name + '</div>');
+	result.append(result.title);
+
+	result.control = $('<div class="scale-control"/>');
+	result.append(result.control);
+
+	result.slider = $('<input class="scale-slider" type="range"/>');
+	result.slider.attr({
+		min : 0,
+		max : 255,
+		step : 1
+	});
+	if(_name == "Alpha")
+	{
+		result.slider.attr({
+			min : 0,
+			max : 1,
+			step : 0.01
+		});
+	}
+	result.slider.val(this.popup.values[_name]);
+	result.control.append(result.slider);
+
+	result.label = $('<div class="scale-label title-font-big font-bold font-color-title">' + this.popup.values[_name] + '</div>');
+	result.append(result.label);
+
+	result.on("change", function ()
+	{
+		self.popup.values[_name] = parseFloat(result.slider.val());
+		result.label.text('' + self.popup.values[_name]);
+		self.updateColorInPopup()
+	});
+}
+
+ColorPickerSetting.prototype.cloneSettingsToFrom = function ( _destination, _source)
+{
+	_destination.Red = _source.Red;
+	_destination.Green = _source.Green;
+	_destination.Blue = _source.Blue;
+	_destination.Alpha = _source.Alpha;
+}
+
+ColorPickerSetting.prototype.getRGBA = function( _values )
+{
+	var result = "rgba(";
+	result += _values.Red + ", ";
+	result += _values.Green + ", ";
+	result += _values.Blue + ", ";
+	result += _values.Alpha + ");";
+	return result
+}
```

Comment:
Not sure why these functions are static

Reply:
There's only ever going to be these entries. It shows you exactly what's happening. No need to iterate over the object here-